lib/src/main/kotlin/com/swmansion/starknet/data/types/Execution.kt

```diff
@@ -3,11 +3,12 @@
 package com.swmansion.starknet.data.types
 
 import com.swmansion.starknet.data.selectorFromName
+import com.swmansion.starknet.data.types.conversions.ConvertibleToCalldata
 import com.swmansion.starknet.data.types.transactions.InvokeTransaction
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
-typealias Calldata = List<Felt>
+typealias Calldata = List<ConvertibleToCalldata>
```

Comment:
Hmmm this might little be a bit misleading. I think we are missing an additional type.
Calldata - calldata as accepted on starknet - list of felts
CallArguments (might need a better name?) - list of arguments for call - list of ConvertibleToCalldata

What do you think?

Reply:
Problem is `CallArgument` doesn't make much sense in many places where `Calldata` is currently used e.g. in `DeployAccountTransactionPayload` transaction, UDC deployer etc.

We could of course decide to continue using `Calldata` there, but we kind of loose all benefits of defining this `ConvertibleToCalldata` if it's only going to be used in InvokeTransactions.

Reply:
`DeployAccountTransactionPayload` would use `Calldata`, not `CallArguments`. I want `CallArgument` to be high level interface for creating calladata.

Reply:
So you want `CallArgument` to only be usable when creating `Call` instances?

Reply:
Actually I like this approach, we could not introduce all these changes added in the PR then, and settle on some kind of conversion for `Call` keeping the base type there `List<Felt>` in other places. 

Perhaps having a factory method `Call.fromCallArguments` or something like that that accepts this higher level interface? (ideally we'd have a constructors with both `List<Felt>` and `List<ConvertibleToCalldata>` but we can have that in jvm.)

Reply:
What if we used Array for calldata and List for arguments? Then we could support both.

Reply:
I don't like this approach, forcing users to use different data types. I believe more expected solution for Java users to this situation is to use static methods.